### PR TITLE
Move code in `util` package to `_internal`

### PR DESCRIPTION
**Pull request by shsms**
There was just a single function `cancel_and_await` that was only for internal use.

### CODE REVIEW

**Review by matthias-wende-frequenz — DISMISSED** [review]

File: src/frequenz/sdk/_internal/__init__.py [hunk 1]
```diff
@@ -3,7 +3,8 @@
 
 """Utility types and functions for internal use."""
 
+from .asyncio import cancel_and_await
 from .singleton_meta import SingletonMeta
 
 # Explicitly declare the public API.
-__all__ = ["SingletonMeta"]
+__all__ = ["SingletonMeta", "cancel_and_await"]
```
Comment (ela-kotulska-frequenz) [inline src/frequenz/sdk/_internal/__init__.py]: This module has generic name `_internal`. 

Because of that separation into submodule `asyncio` is fine.

Comment (shsms) [reply src/frequenz/sdk/_internal/__init__.py]: it is in a submodule already.  are you saying we shouldn't export reexport the function from `_internal`?

Comment (shsms) [reply src/frequenz/sdk/_internal/__init__.py]: I've updated the PR with that assumption.

**Review by ela-kotulska-frequenz — APPROVED** [review]

File: src/frequenz/sdk/_internal/__init__.py (hunk 1 above)
Comment (ela-kotulska-frequenz) [reply src/frequenz/sdk/_internal/__init__.py]: Yes that what I meant :)

**Comment by leandro-lucarella-frequenz** [comment]
The idea was that this was public as a way to provide SDK users with useful tools, I really don't see why we have to keep this internal. For example `await_and_cancel()` could be useful for the channels repo. Of course we can't use it there because it would be a circular dependency, but I thought we might want to extract the `util` package and make it a separate library like we did with channels.

For example, I'm looking into https://github.com/frequenz-floss/frequenz-sdk-python/discussions/197, and that's another clear candidate for this, and can be helpful for channels too, as we also have this kind of objects there (`Merge`, `MergeNamed` and `Select`).

If there is no big opposition, I will move the package back to being public, and add the *background service* class in `util.asyncio` too, I think it fits there.

**Pull request merged**